applications/serial_lte_modem/src/slm_util.c

```diff
@@ -93,32 +94,67 @@ int slm_util_at_scanf(const char *cmd, const char *fmt, ...)
 	return ret;
 }
 
-int slm_util_at_cmd_fwd_from_cb(char *buf, size_t len, char *at_cmd)
+static int terminate_at_response(char *buf, size_t len,
+				 const char *termination_str, int success_ret)
 {
-/* This gets called by interception callbacks invoked by nrf_modem_at_cmd().
+	len -= strlen(buf);
+	const int printf_ret = snprintf(buf + strlen(buf), len, "%s", termination_str);
+
+	if (printf_ret >= len) {
+		LOG_ERR("%u bytes of the AT response were truncated."
+			" The buffer needs to be made bigger.", printf_ret - len + 1);
+		return -NRF_E2BIG;
+	}
+	return success_ret;
+}
+
+int slm_util_at_cmd_no_intercept(char *buf, size_t len, const char *at_cmd)
+{
+	int ret;
+	char second_line[strlen("+CMx ERROR: xxxx")];
+	char format_str[24];
+
+	/* Construct the format string to read two response lines (into buf and second_line). */
+	ret = snprintf(format_str, sizeof(format_str),
+		       "%%%u[^\r\n]\r\n%%%u[^\r\n]", len - 1, sizeof(second_line) - 1);
+	assert(ret < sizeof(format_str));
+
+	LOG_DBG("Forwarding \"%s\" to the modem.", at_cmd);
+
+/* This function gets called by interception callbacks invoked by nrf_modem_at_cmd().
  * Here nrf_modem_at_cmd() must be bypassed because it would otherwise call
  * the same interception callbacks and infinite recursion would happen.
  */
-#undef nrf_modem_at_printf
-	const int ret = nrf_modem_at_printf("%s", at_cmd);
+#undef nrf_modem_at_scanf
+	const int line_count = nrf_modem_at_scanf(at_cmd, format_str, buf, second_line);
```

Comment:
What happens if the response is 3 lines or more? If that is not the case, is then the OK is not in the resp_str and whole response is overwritten with ERROR?

Reply:
Correct.